### ISSUE TITLE
Feature Request: Enhance Whiteboard with Common Basic Tools

### ISSUE THREAD

**Issue by gjaekel** [xwiki-labs/cryptpad#679]
At the current state, even with V4.0, the Whiteboard have -- well, let's say -- "still much potential to enhance".

_Please_ add in near-time some basic tools like scalable and movable geometric objects like a rectangle or ellipse. 

Since about one year I would **really** like to introduce the CryptPad Suit, but an important user story is to have a collaboration to for interactive project plannings. This will need to have a board with a "fixed" background layer -- the "image" tool can be used here -- and to write down and move some "sticky notes" on that.

In the other hand, the other apps of the CryptPad suite are well featured. From that, I would like to offer the whole suite and don't like to add another whiteboard app.

**Comment by TheNomad11** [comment]
Inserting  video and audio files would also be useful, especially in education settings

**Comment by ansuz** [comment]
I agree that there is considerable room for improvement in the whiteboard app, however, our current roadmap is sufficient to occupy our team full time at least until early 2022. If any of the large public institutions using CryptPad for remote collaboration or education volunteered to fund this development we could find a way to implement these improvements sooner. Otherwise, we may consider these improvements as the topic for a future grant proposal.

**Comment by gjaekel** [comment]
Thank you at least for reply. Unfortunately, we look mostly for a shared whiteboard as an add-on for our A/V meeting service.

**Comment by unfa** [comment]
I agree that the whiteboard could use some work.
What I'm missing the most is an infinite canvas and navigation.
Having one screen to draw on is very limiting, especially if you can't zoom in to "make more room" that way.
I'd love to be able to zoom out and drag objects far away, then zoom back in and be able to see a mini-map showing me how big the canvas is, and what part of it am I looking at.
At that point it'd already allow teams to collaboratively plan designs, processes and other things - right now there's simply not enough room.

**Comment by zuzanna-maria** [comment]
Closing this feature request as most if not all of these features are now available in our Diagram app, and as such we are not planning to implement them in Whiteboard.
